pkg/object/statussynccontroller/statussynccontroller.go

```diff
@@ -235,8 +235,7 @@ func (ssc *StatusSyncController) syncStatusToCluster(statusUnits map[string]*sta
 	if ssc.lastSyncStatusUnits != nil {
 		kvs := make(map[string]*string)
 		for k := range ssc.lastSyncStatusUnits {
-			if su, exists := statusUnits[k]; !exists {
-				k = su.clusterKey(ssc.superSpec.Super().Cluster().Layout())
+			if _, exists := statusUnits[k]; !exists {
```

Comment:
```go
		for k, su := range ssc.lastSyncStatusUnits {
			if _, exists := statusUnits[k]; !exists {
				key := su.clusterKey(ssc.superSpec.Super().Cluster().Layout())
				kvs[key] = nil
			}
		}
```

It seems this kind of change will fix the problem totally, how about testing and applying it?